dlt/destinations/impl/sqlalchemy/sqlalchemy_job_client.py

```diff
@@ -0,0 +1,503 @@
+from typing import Iterable, Optional, Type, Dict, Any, Iterator, Sequence, List, Tuple, IO
+from types import TracebackType
+from contextlib import suppress
+import inspect
+import math
+
+import sqlalchemy as sa
+from sqlalchemy.sql import sqltypes
+
+from dlt.common import logger
+from dlt.common import pendulum
+from dlt.common.exceptions import TerminalValueError
+from dlt.common.destination.reference import (
+    JobClientBase,
+    LoadJob,
+    RunnableLoadJob,
+    StorageSchemaInfo,
+    StateInfo,
+)
+from dlt.destinations.job_client_impl import SqlJobClientBase
+from dlt.common.destination.capabilities import DestinationCapabilitiesContext
+from dlt.common.schema import Schema, TTableSchema, TColumnSchema, TSchemaTables
+from dlt.common.schema.typing import TColumnType, TTableFormat, TTableSchemaColumns
+from dlt.common.schema.utils import pipeline_state_table, normalize_table_identifiers
+from dlt.common.storages import FileStorage
+from dlt.common.json import json, PY_DATETIME_DECODERS
+from dlt.destinations.exceptions import DatabaseUndefinedRelation
+
+
+# from dlt.destinations.impl.sqlalchemy.sql_client import SqlalchemyClient
+from dlt.destinations.impl.sqlalchemy.db_api_client import SqlalchemyClient
+from dlt.destinations.impl.sqlalchemy.configuration import SqlalchemyClientConfiguration
+
+
+class SqlaTypeMapper:
+    # TODO: May be merged with TypeMapper as a generic
+    def __init__(
+        self, capabilities: DestinationCapabilitiesContext, dialect: sa.engine.Dialect
+    ) -> None:
+        self.capabilities = capabilities
+        self.dialect = dialect
+
+    def _db_integer_type(self, precision: Optional[int]) -> sa.types.TypeEngine:
+        if precision is None:
+            return sa.BigInteger()
+        elif precision <= 16:
+            return sa.SmallInteger()
+        elif precision <= 32:
+            return sa.Integer()
+        elif precision <= 64:
+            return sa.BigInteger()
+        raise TerminalValueError(f"Unsupported precision for integer type: {precision}")
+
+    def _create_date_time_type(self, sc_t: str, precision: Optional[int]) -> sa.types.TypeEngine:
+        """Use the dialect specific datetime/time type if possible since the generic type doesn't accept precision argument"""
+        precision = precision if precision is not None else self.capabilities.timestamp_precision
+        if sc_t == "timestamp":
```

Comment:
we support datetimes without timezones now see [#1492](https://github.com/dlt-hub/dlt/pull/1669)